internal/commands/build/compose.go

```diff
@@ -0,0 +1,70 @@
+package build
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/docker/app/types"
+	"github.com/docker/buildx/build"
+	"github.com/docker/cli/cli/compose/loader"
+	compose "github.com/docker/cli/cli/compose/types"
+)
+
+// parseCompose do parse app compose file and extract buildx Options
+// We don't rely on bake's ReadTargets + TargetsToBuildOpt here as we have to skip environment variable interpolation
+func parseCompose(app *types.App, options buildOptions) (map[string]build.Options, error) {
+	parsed, err := loader.ParseYAML(app.Composes()[0])
+	if err != nil {
+		return nil, err
+	}
+
+	services, err := Load(parsed)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse compose file: %s", err)
+	}
+
+	opts := map[string]build.Options{}
+	for _, service := range services {
+		if service.Build == nil {
+			continue
+		}
+
+		var tags []string
+		if service.Image != nil {
```

Comment:
Is it possible here to have a non-nil `Image` but empty string? 🤔 
Like if the compose file declares
```yaml
services:
  myservice:
    image: ""
```

Reply:
Just checked, and `docker-compose` allows this and just ignore. Weird :P